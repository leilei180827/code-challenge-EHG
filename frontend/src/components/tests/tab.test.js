import React from 'react';
import renderer from 'react-test-renderer';
import Tab from '../tab';

test('tab render correctly', () => {
  const handleImageTypeTab = (type) => {
  }
  const component = renderer.create(
    <Tab handleImageTypeTab={handleImageTypeTab}></Tab>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
