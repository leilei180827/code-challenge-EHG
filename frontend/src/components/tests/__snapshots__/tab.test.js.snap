// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tab render correctly 1`] = `
<div
  className="ant-tabs ant-tabs-top ant-tabs-centered"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <div
    className="ant-tabs-nav"
    onKeyDown={[Function]}
    role="tablist"
  >
    <div
      className="ant-tabs-nav-wrap"
    >
      <div
        className="ant-tabs-nav-list"
        style={
          Object {
            "transform": "translate(0px, 0px)",
            "transition": undefined,
          }
        }
      >
        <div
          className="ant-tabs-tab"
          onClick={[Function]}
          style={
            Object {
              "marginRight": undefined,
            }
          }
        >
          <div
            aria-controls={null}
            aria-selected={false}
            className="ant-tabs-tab-btn"
            id={null}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="tab"
            tabIndex={0}
          >
            default
          </div>
        </div>
        <div
          className="ant-tabs-tab"
          onClick={[Function]}
          style={
            Object {
              "marginRight": undefined,
            }
          }
        >
          <div
            aria-controls={null}
            aria-selected={false}
            className="ant-tabs-tab-btn"
            id={null}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="tab"
            tabIndex={0}
          >
            rgb
          </div>
        </div>
        <div
          className="ant-tabs-tab"
          onClick={[Function]}
          style={
            Object {
              "marginRight": undefined,
            }
          }
        >
          <div
            aria-controls={null}
            aria-selected={false}
            className="ant-tabs-tab-btn"
            id={null}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="tab"
            tabIndex={0}
          >
            gray
          </div>
        </div>
        <div
          className="ant-tabs-tab"
          onClick={[Function]}
          style={
            Object {
              "marginRight": undefined,
            }
          }
        >
          <div
            aria-controls={null}
            aria-selected={false}
            className="ant-tabs-tab-btn"
            id={null}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="tab"
            tabIndex={0}
          >
            hsv
          </div>
        </div>
        <div
          className="ant-tabs-tab"
          onClick={[Function]}
          style={
            Object {
              "marginRight": undefined,
            }
          }
        >
          <div
            aria-controls={null}
            aria-selected={false}
            className="ant-tabs-tab-btn"
            id={null}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="tab"
            tabIndex={0}
          >
            gradual-rgb
          </div>
        </div>
        <div
          className="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
        />
      </div>
    </div>
    <div
      className="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
    >
      <button
        aria-controls="null-more-popup"
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-hidden="true"
        className="ant-tabs-nav-more"
        id="null-more"
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "marginRight": undefined,
            "order": 1,
            "visibility": "hidden",
          }
        }
        tabIndex={-1}
        type="button"
      >
        <span
          aria-label="ellipsis"
          className="anticon anticon-ellipsis"
          role="img"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="ellipsis"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    className="ant-tabs-content-holder"
  >
    <div
      className="ant-tabs-content ant-tabs-content-top"
      style={null}
    >
      <div
        aria-hidden={true}
        aria-labelledby={null}
        className="ant-tabs-tabpane"
        id={null}
        role="tabpanel"
        style={
          Object {
            "display": "none",
          }
        }
        tabIndex={-1}
      />
      <div
        aria-hidden={true}
        aria-labelledby={null}
        className="ant-tabs-tabpane"
        id={null}
        role="tabpanel"
        style={
          Object {
            "display": "none",
          }
        }
        tabIndex={-1}
      />
      <div
        aria-hidden={true}
        aria-labelledby={null}
        className="ant-tabs-tabpane"
        id={null}
        role="tabpanel"
        style={
          Object {
            "display": "none",
          }
        }
        tabIndex={-1}
      />
      <div
        aria-hidden={true}
        aria-labelledby={null}
        className="ant-tabs-tabpane"
        id={null}
        role="tabpanel"
        style={
          Object {
            "display": "none",
          }
        }
        tabIndex={-1}
      />
      <div
        aria-hidden={true}
        aria-labelledby={null}
        className="ant-tabs-tabpane"
        id={null}
        role="tabpanel"
        style={
          Object {
            "display": "none",
          }
        }
        tabIndex={-1}
      />
    </div>
  </div>
</div>
`;
